api/lib/application/authentication/index.js

```diff
@@ -43,13 +43,7 @@ exports.register = async (server) => {
             token_type_hint: 'access_token',
           }),
           failAction: (request, h) => {
-            const errorHttpStatusCode = 400;
-            const jsonApiError = new JSONAPIError({
-              status: errorHttpStatusCode.toString(),
-              title: 'Bad request',
-              detail: 'The server could not understand the request due to invalid syntax.',
-            });
-            return h.response(jsonApiError).code(errorHttpStatusCode).takeover();
+            return sendJsonApiError(new BadRequestError('The server could not understand the request due to invalid token.'), h);
```

Comment:
Remarque mineure (ne pas implémenter si le refacto prend plus de 15 minutes)

Le résultat de ce code est que le serveur répond au client 
- qu'une erreur est survenue` .code(httpError.status)`
- que le payload au format (au JSONAAPI) contient le détail de l'erreur

En gros, elle sérialise l'objet `httpError` puis répond au client avec. Comme le format du payload est toujours du JSONAPI, je me demande s'il est utile de l'indiquer. Ce qui me semble important, c'est que le développeur sache que cette fonction fait bien le `h.response.takeover()` sans devoir l'ouvrir pour vérifier

Je suggère de reommer `sendJsonApiError`par `replyWithError' ou `serializeErrorAndReplyToClient`
```suggestion
            return serializeErrorAndReplyToClient(new BadRequestError('The server could not understand the request due to invalid token.'), h);
```

Reply:
Oui, construire une `response` avec une `JSONAPIError` et le HTTP statusCode (`code`) correspondant à celui de la class passée en paramètre (`status`).

Pour le nom, dans l'absolue, les send et reply ne sont pas adéquats. C'est juste le `return`qui "transmettra" la `response`...

Mentionner la `response`contruite avec une JSONAPIError et le statusCode ... je ne trouve pas l'inspiration :)

Reply:
Ca roule, laisse comme ça alors